abci/check_tx.go

```diff
@@ -0,0 +1,186 @@
+package abci
+
+import (
+	"context"
+	"fmt"
+
+	cometabci "github.com/cometbft/cometbft/abci/types"
+	log "github.com/cometbft/cometbft/libs/log"
+	tmproto "github.com/cometbft/cometbft/proto/tendermint/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/skip-mev/pob/mempool"
+)
+
+type (
+	// CheckTxHandler is a wrapper around baseapp's CheckTx method that allows us to
+	// verify bid transactions against the latest committed state. All other transactions
+	// are executed normally using base app's CheckTx. This defines all of the
+	// dependencies that are required to verify a bid transaction.
+	CheckTxHandler struct {
+		// baseApp is utilized to retrieve the latest committed state and to call
+		// baseapp's CheckTx method.
+		baseApp BaseApp
+
+		// txDecoder is utilized to decode transactions to determine if they are
+		// bid transactions.
+		txDecoder sdk.TxDecoder
+
+		// mempool is utilized to retrieve the bid info of a transaction and to
+		// insert a transaction into the application-side mempool.
+		mempool CheckTxMempool
+
+		// anteHandler is utilized to verify the bid transaction against the latest
+		// committed state.
+		anteHandler sdk.AnteHandler
+
+		// chainID is the chain ID of the blockchain.
+		chainID string
+	}
+
+	// CheckTx is baseapp's CheckTx method that checks the validity of a
+	// transaction.
+	CheckTx func(cometabci.RequestCheckTx) cometabci.ResponseCheckTx
```

Comment:
This is already defined in the SDK (types/abci.go). Why define it again?

Reply:
I didnt see it in the `v0.47.2` release (https://github.com/cosmos/cosmos-sdk/blob/v0.47.2/types/abci.go) so redefined it here.

Reply:
Ahh you're right! Ok, no biggie